weave-js/src/errors.ts

```diff
@@ -0,0 +1,57 @@
+import {WeaveApp} from './weave';
+
+export class UseNodeValueServerExecutionError extends Error {}
+
+export const errorToPayload = (
+  error: Error,
+  weave?: WeaveApp
+): {[prop: string]: any} => {
+  try {
+    return {
+      errorMessage: trimString(error.message),
+      errorName: trimString(error.name),
+      errorStack: trimString(error.stack),
+      hashedErrorMessage: hashStr(error.message),
+      hashedErrorStack: hashStr(error.stack),
+      // segment analytics creates a context_page_url which is supposed to
+      // be the current page, but in practice, this seems to be broken
+      // about 15% of the time. Adding this manually for now.
+      windowLocationURL: trimString(window.location.href),
+      weaveContext: weave?.client.debugMeta(),
+      isServerError: error instanceof UseNodeValueServerExecutionError,
+    };
+  } catch (e) {
+    // If we fail to serialize the error, just return an empty object.
+    console.error('Failed to serialize error', e);
+    return {};
+  }
+};
+
+const MAX_CONFIG_STRING_LENGTH = 2500;
+export const trimString = (str?: string) => {
+  if (str == null) {
+    return '';
+  }
+  if (str.length > MAX_CONFIG_STRING_LENGTH) {
+    return str.slice(0, MAX_CONFIG_STRING_LENGTH) + '...';
```

Comment:
Does something actually have MAX_CONFIG_STRING_LENGTH as a limit? Because the ellipsis would make it longer than that.

Reply:
Not that i am aware, but removed the append anyhow (btw this is factored out of core)